packages/astro/src/assets/utils/svg.ts

```diff
@@ -0,0 +1,54 @@
+import { parse, renderSync } from 'ultrahtml';
+import type { ImageMetadata } from '../types.js';
+import type { SvgComponentProps } from '../runtime.js';
+type SvgAttributes = Record<string, any>;
+
+/**
+ * Some attributes required for `image/svg+xml` are irrelevant when inlined ina `text/html` document. We can save a few bytes by dropping them.
+ */
+const ATTRS_TO_DROP = ['xmlns', 'xmlns:xlink', 'version'];
+const DEFAULT_ATTRS: SvgAttributes = { role: 'img' };
+
+function dropAttributes(attributes: SvgAttributes) {
+	for (const attr of ATTRS_TO_DROP) {
+		delete attributes[attr];
+	}
+
+	return attributes;
+}
+
+export function normalizeProps(attributes: SvgAttributes, { size, ...props }: SvgAttributes) {
+	if (size) {
+		props.height = size;
+		props.width = size;
+	}
```

Comment:
Looks like `size` will override `width` and `height` attributes if present. I think `size` should only apply when they are `undefined`.

Reply:
Updated! Will add a test case for this as well!